drivers/dma/dma_mcux_lpc.c

```diff
@@ -193,39 +200,110 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 				config->block_count);
 			return -EINVAL;
 		}
-		int i = 0;
-		dma_transfer_config_t *next_transfer;
+
+		/* Allocate the descriptor table structures if needed */
+		if (data->dma_descriptor_table == NULL) {
+			data->dma_descriptor_table = k_malloc(CONFIG_DMA_LINK_QUEUE_SIZE *
```

Comment:
I did not see a k_free, it is better to free them when stoping transfer.

Reply:
I will add this in the dma_stop function.